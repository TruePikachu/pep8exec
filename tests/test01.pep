	; Program to compute squares for 0..255
	LDA	0,i		; A = 0
	STA	counter,d	; counter = 0
loop:	LDA	counter,d	; A = counter
	LDX	counter,d	; X = counter
	CALL	Multiply	; X:A = A*X
	DECO	counter,d	; CON <- counter
	STRO	delim,d		; CON <- "^2="
	LDX	counter,d	; CON <- A,'\n'
	STA	counter,d	; ...
	DECO	counter,d	; ...
	CHARO	'\n',i		; ...
	ADDX	1,i		; counter++
	STX	counter,d	; ...
	CPX	256,i		; Loop if counter!=256
	BRNE	loop		; ...
	STOP

counter:.WORD	0
delim:	.ASCII	"^2=\x00"

;;; PROCEDURE Multiply
;;; X:A <- A*X
Multiply:SUBSP	10,i
	STX	0,s	; 0:  WORD n1 = X
	LDX	0,i	; X = 0
	STX	2,s	; 2: DWORD n2 = 0:A
	STA	4,s	; ...
	STX	6,s	; 6: DWORD res = 0
	STX	8,s	; ...
Mloop:	LDX	0,s	; X = n1&1
	ANDX	1,i	; ...
	BREQ	MNoAdd	; Jump if bit is clear
	LDX	8,s	; res.L += n2,L
	ADDX	4,s	; ...
	STX	8,s	; ...
	BRC	MCar	; Jump if overflow
	LDX	6,s	; res.H += n2.H
	ADDX	2,s	; ...
	STX	6,s	; ...
	BR	MNoAdd	; Jump to merge
MCar:	LDX	6,s	; res.H += n2.H + 1
	ADDX	1,i	; ...
	ADDX	2,s	; ...
	STX	6,s	; ...
MNoAdd:	LDX	0,s	; n1 /= 2
	ASRX		; ...
	STX	0,s	; ...
	LDA	4,s	; X:A = n2
	LDX	2,s	; ...
	ASLX		; X<<=1
	ASLA		; A<<=1
	BRC	MSh1	; X++ if carry
	SUBX	1,i	; ...
MSh1:	ADDX	1,i	; ...
	STX	2,s	; n2 = X:A
	STA	4,s	; ...
	LDX	0,s	; X = n1
	BRNE	Mloop	; Loop if !=0
	LDX	6,s	; X:A = res
	LDA	8,s	; ...
	ADDSP	10,i
	RET0
;;; END PROCEDURE
	.END
