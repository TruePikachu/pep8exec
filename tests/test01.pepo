C0 00 00 E1 00 2E C1 00 2E C9 00 2E 16 00 34 39
00 2E 41 00 30 C9 00 2E E1 00 2E 39 00 2E 50 00
0A 78 00 01 E9 00 2E B8 01 00 0C 00 06 00 00 00
5E 32 3D 00 68 00 0A EB 00 00 C8 00 00 EB 00 02
E3 00 04 EB 00 06 EB 00 08 CB 00 00 98 00 01 0A
00 76 CB 00 08 7B 00 04 EB 00 08 14 00 6A CB 00
06 7B 00 02 EB 00 06 04 00 76 CB 00 06 78 00 01
7B 00 02 EB 00 06 CB 00 00 1F EB 00 00 C3 00 04
CB 00 02 1D 1C 14 00 8B 88 00 01 78 00 01 EB 00
02 E3 00 04 CB 00 00 0C 00 49 CB 00 06 C3 00 08
60 00 0A 58 zz
