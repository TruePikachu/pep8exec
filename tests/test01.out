0^2=0
1^2=1
2^2=4
3^2=9
4^2=16
5^2=25
6^2=36
7^2=49
8^2=64
9^2=81
10^2=100
11^2=121
12^2=144
13^2=169
14^2=196
15^2=225
16^2=256
17^2=289
18^2=324
19^2=361
20^2=400
21^2=441
22^2=484
23^2=529
24^2=576
25^2=625
26^2=676
27^2=729
28^2=784
29^2=841
30^2=900
31^2=961
32^2=1024
33^2=1089
34^2=1156
35^2=1225
36^2=1296
37^2=1369
38^2=1444
39^2=1521
40^2=1600
41^2=1681
42^2=1764
43^2=1849
44^2=1936
45^2=2025
46^2=2116
47^2=2209
48^2=2304
49^2=2401
50^2=2500
51^2=2601
52^2=2704
53^2=2809
54^2=2916
55^2=3025
56^2=3136
57^2=3249
58^2=3364
59^2=3481
60^2=3600
61^2=3721
62^2=3844
63^2=3969
64^2=4096
65^2=4225
66^2=4356
67^2=4489
68^2=4624
69^2=4761
70^2=4900
71^2=5041
72^2=5184
73^2=5329
74^2=5476
75^2=5625
76^2=5776
77^2=5929
78^2=6084
79^2=6241
80^2=6400
81^2=6561
82^2=6724
83^2=6889
84^2=7056
85^2=7225
86^2=7396
87^2=7569
88^2=7744
89^2=7921
90^2=8100
91^2=8281
92^2=8464
93^2=8649
94^2=8836
95^2=9025
96^2=9216
97^2=9409
98^2=9604
99^2=9801
100^2=10000
101^2=10201
102^2=10404
103^2=10609
104^2=10816
105^2=11025
106^2=11236
107^2=11449
108^2=11664
109^2=11881
110^2=12100
111^2=12321
112^2=12544
113^2=12769
114^2=12996
115^2=13225
116^2=13456
117^2=13689
118^2=13924
119^2=14161
120^2=14400
121^2=14641
122^2=14884
123^2=15129
124^2=15376
125^2=15625
126^2=15876
127^2=16129
128^2=16384
129^2=16641
130^2=16900
131^2=17161
132^2=17424
133^2=17689
134^2=17956
135^2=18225
136^2=18496
137^2=18769
138^2=19044
139^2=19321
140^2=19600
141^2=19881
142^2=20164
143^2=20449
144^2=20736
145^2=21025
146^2=21316
147^2=21609
148^2=21904
149^2=22201
150^2=22500
151^2=22801
152^2=23104
153^2=23409
154^2=23716
155^2=24025
156^2=24336
157^2=24649
158^2=24964
159^2=25281
160^2=25600
161^2=25921
162^2=26244
163^2=26569
164^2=26896
165^2=27225
166^2=27556
167^2=27889
168^2=28224
169^2=28561
170^2=28900
171^2=29241
172^2=29584
173^2=29929
174^2=30276
175^2=30625
176^2=30976
177^2=31329
178^2=31684
179^2=32041
180^2=32400
181^2=32761
182^2=-32412
183^2=-32047
184^2=-31680
185^2=-31311
186^2=-30940
187^2=-30567
188^2=-30192
189^2=-29815
190^2=-29436
191^2=-29055
192^2=-28672
193^2=-28287
194^2=-27900
195^2=-27511
196^2=-27120
197^2=-26727
198^2=-26332
199^2=-25935
200^2=-25536
201^2=-25135
202^2=-24732
203^2=-24327
204^2=-23920
205^2=-23511
206^2=-23100
207^2=-22687
208^2=-22272
209^2=-21855
210^2=-21436
211^2=-21015
212^2=-20592
213^2=-20167
214^2=-19740
215^2=-19311
216^2=-18880
217^2=-18447
218^2=-18012
219^2=-17575
220^2=-17136
221^2=-16695
222^2=-16252
223^2=-15807
224^2=-15360
225^2=-14911
226^2=-14460
227^2=-14007
228^2=-13552
229^2=-13095
230^2=-12636
231^2=-12175
232^2=-11712
233^2=-11247
234^2=-10780
235^2=-10311
236^2=-9840
237^2=-9367
238^2=-8892
239^2=-8415
240^2=-7936
241^2=-7455
242^2=-6972
243^2=-6487
244^2=-6000
245^2=-5511
246^2=-5020
247^2=-4527
248^2=-4032
249^2=-3535
250^2=-3036
251^2=-2535
252^2=-2032
253^2=-1527
254^2=-1020
255^2=-511
